docs/types/buffer.md

```diff
@@ -0,0 +1,61 @@
+# Buffers
+
+Buffers are what are used to temporarily store log entries until they can be flushed to their final destination.
+
+There are two types of buffers: `memory` buffers and `disk` buffers.
+
+## Memory Buffers
+
+Memory buffers keep log entries in memory until they are flushed, which makes them very fast. However, because
+entries are only stored in memory, they will be lost if the agent is shut down uncleanly. If the agent is shut down
+cleanly, they will be saved to the agent's database.
+
+### Memory Buffer Configuration
+
+Memory buffers are configured by setting the `type` field of the `buffer` block on an output to `memory`. The only other
+configurable field is `max_entries`, which is maximum number of entries that will be held in memory before blocking and
```

Comment:
This doesn't have to happen in this pr, but we should probably offer another config option to control the behave when full. For example:

`on_full: block` // default
`on_full: drop_oldest`

Reply:
Agreed. I'll add that to the list of followup features